### PR TITLE
build!: upgrade all packages + root to latest Hardhat version and ethers v6

### PR DESCRIPTION
**Pull request by CJ42**
# What does this PR introduce?

## 📦 Build

> **Note:** the tests for `@lukso/lsp17-extensions` have been skipped due to an issue related to the deployment of NickFactory. See this opened issue on Hardhat repository for reference: https://github.com/NomicFoundation/hardhat/issues/4939

Upgrade Hardhat and ethers to latest version, including ethers v6.

- Upgrade dependencies to latest, including:
    - `hardhat` to latest `v2.20.1`
    - `@nomicfoundation/hardhat-toolbox` to `v4.0.0`
    
- Refactor all the scripts for building, tasks, testing and deployment to ethers v6

### PR Checklist

- [ ] Wrote Tests
- [ ] Wrote & Generated Documentation (readme/natspec/dodoc)
- [ ] Ran `npm run lint` && `npm run lint:solidity` (solhint)
- [ ] Ran `npm run format` (prettier)
- [ ] Ran `npm run build`
- [ ] Ran `npm run test`

### CODE REVIEW

**Comment by CJ42** [comment]
Referencing this issue on Hardhat for the blocker on LSP17 Extensions contracts tests and the Create2Factory not deploying: https://github.com/NomicFoundation/hardhat/issues/4939

File: tests/LSP17Extensions/helpers/UserOp.ts [hunk 1]
```diff
@@ -41,19 +42,19 @@ export function packUserOp(op: UserOperation, forSignature = true): string {
       [
         op.sender,
         op.nonce,
-        keccak256(op.initCode),
-        keccak256(op.callData),
+        keccak256(op.initCode as BytesLike),
```
Comment (richtera) [inline tests/LSP17Extensions/helpers/UserOp.ts:45]: Here you're already doing the import correctly.

**Review by richtera — APPROVED** [review]
This looks ok, but I would remove the ethers wholesale import and specify each function you need.

File: packages/lsp14-contracts/hardhat.config.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ import { resolve } from 'path';
 import '@nomicfoundation/hardhat-toolbox';
 
 // additional hardhat plugins
-import 'hardhat-packager';
+// import 'hardhat-packager';
```
[the remark this answers is not in the document]
Comment (CJ42) [reply packages/lsp14-contracts/hardhat.config.ts:19]: good catch indeed. Don't know why it is still commented.

**Review by b00ste — APPROVED** [review]

**Review by skimaharvey — APPROVED** [review]
Awesome 🙏

**Pull request merged**